tensorboard/components/tf_trace_viewer/tf-trace-viewer.html

```diff
@@ -97,44 +113,229 @@
           var components = parts[i].split('=');
           if (components[0] == "trace_data_url") {
             this.traceDataUrl = decodeURIComponent(components[1]);
-            break;
+          } else if (components[0] == "is_streaming") {
+            this._isStreaming = components[1] === 'true';
           }
         }
       }
 
+      if (!this.traceDataUrl) {
+        this._displayOverlay("Trace data URL is not provided.", "Trace Viewer");
+        return null;
+      }
       this._throbber.className = "active";
+
       this._loadTrace();
     },
+
+    // Something has changed, so consider reloading the data:
+    //   - if we have zoomed in enough to need more details.
+    //   - if we have scrolled too close to missing data regions
+    // We ensure there's only ever one request in flight.
     _loadTrace : function() {
-      if (!this.traceDataUrl) {
-        this._displayOverlay("Trace data URL is not provided.", "Trace Viewer");
-        return null;
+      if (!this._isStreaming) {
+        // Send HTTP request to get the trace data.
+        var req = new XMLHttpRequest();
+        var is_binary = / [.] gz$ /.test(this.traceDataUrl) ||
+                        / [.] zip$ /.test(this.traceDataUrl);
+        req.overrideMimeType('text/plain; charset=x-user-defined');
```

Comment:
Why is the mime type being set to `text/plain; charset=x-user-defined`?

Reply:
ditto, I don't know. probably not used.

Reply:
done .removed.